### PR TITLE
docs: update CONTRIBUTING.md to split frontend/backend and add emulator info

### LINKED ISSUE #1427: Clarify contributing.md for frontend work
Review CONTRIBUTING.md for those considering front end updates. Are the instructions clear? Is there enough information?

### PR DESCRIPTION
**Pull request by jess-lowe**
Updated CONTRIBUTING.md to improve the contributor experience by separating frontend and backend development instructions and providing information on how to use the website emulator for local testing without GCP credentials.
- Split "Contributing code" into "Backend development" and "Frontend development".
- Add instructions for non-internal contributors to use `make run-website-emulator`.
- Mention adding testcase records to `gcp/website/testdata/osv/` for testing edge cases.
- Fix a minor typo in the API server section.

Closes #1427

---
*PR created automatically by Jules for task [4582872313076138700](https://jules.google.com/task/4582872313076138700) started by @jess-lowe*

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -109,49 +111,63 @@ the environment variable `TESTS_GENERATE=1` and run the tests:
...
-#### UI
+#### Running a local UI instance (maintainers only)
 
 ```shell
 gcloud auth login --update-adc
 make run-website
 ```
 
-#### API
+#### Running a local UI instance (non-internal contributors)
```
Comment (jess-lowe) [inline CONTRIBUTING.md]:
```suggestion
#### Running a local UI instance
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -109,49 +111,63 @@ the environment variable `TESTS_GENERATE=1` and run the tests:
...
 make run-website
 ```
 
-#### API
+#### Running a local UI instance
 
-Running a local instance of the API server requires the path to application
-default credentials. The is required so that the ESP container has credentials
-to download API configuration.
+For non-internal contributors, you can use the website emulator which does
```
Comment (jess-lowe) [inline CONTRIBUTING.md]:
```suggestion
For contributors without access to the GCP project, you can use the website emulator which does
```
